tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -26,20 +28,372 @@ def test_hyperopt_parameters(use_hyperband: bool) -> None:
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) != suggestion
+
+    sampler = TPESampler(prior_weight=0.2, n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) != suggestion
+
+
+def test_sample_relative_n_startup_trial() -> None:
+    warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+
+    study = optuna.create_study()
+    dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 8)]
+
+    trial = frozen_trial_factory(8)
+    # sample_relative returns {} for only 3 observations.
+    sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials[:4]):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) == {}
+    # sample_relative returns some value for only 7 observations.
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert "param-a" in sampler.sample_relative(study, trial, {"param-a": dist}).keys()
+
+
+def test_sample_relative_misc_arguments() -> None:
+    warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+
+    study = optuna.create_study()
+    dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 40)]
+
+    # Prepare a trial and a sample for later checks.
+    trial = frozen_trial_factory(40)
+    sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        suggestion = sampler.sample_relative(study, trial, {"param-a": dist})
+
+    # Test misc. parameters.
+    sampler = TPESampler(n_ei_candidates=13, n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) != suggestion
+
+    sampler = TPESampler(gamma=lambda _: 5, n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) != suggestion
+
+    sampler = TPESampler(
+        weights=lambda n: np.asarray([i ** 2 + 1 for i in range(n)]),
+        n_startup_trials=5,
+        seed=0,
+        multivariate=True,
+    )
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        assert sampler.sample_relative(study, trial, {"param-a": dist}) != suggestion
+
+
+def test_sample_relative_uniform_distributions() -> None:
```

Comment:
Are these introduced because `samplers_tests/test_sampler.py` only cover independent sampling? I'm wondering if we shouldn't work on `samplers_tests/test_sampler.py` instead. It could be a separate issue.

Reply:
Thank you for the insightful comment. Yes, you are correct. I think we can just remove these `test_sample_relative_DISTRIBUTIONS()` without any decrease in test coverage.

Reply:
Thank you for the insightful comment. Actually, we can test `sample_relative` not only for `TPESampler(multivariate=True)` but also for any distributions in `sampelrs_tests/test_samplers.py`. We can work on this in another PR as a followup. 

Here, how about leaving them as it is?